### DISCUSSION TITLE
Could not build with Optix

### DISCUSSION THREAD

**Discussion by popolin522** [glotzerlab/fresnel#261] [Support]
### Description
I tried to build fresnel package using CMake with OptiX support enabled on Windows, but it seems to me Optix support wasn't properly installed. It could be possible that I didn't configure the environment correctly. I tried to do it on WSL before and learned in the hard way that WSL doesn't support Optix, so I am trying to install fresnel with Optix support on powershell.
### Things I did
1. set up a conda environment called `fresnel`
2. follow the [instructions](https://fresnel.readthedocs.io/en/v0.13.7/building.html) and install packages I need (cmake embree git...)
3. download Optix 6.5 (to path C:\ProgramData\NVIDIA Corporation\OptiX SDK 6.5.0\ in my case)
4. run command on powershell `cmake -B build -S . -GNinja -DBUILD_OPTIX=on -DENABLE_EMBREE=ON -DOpTiX_DIR="C:\ProgramData\NVIDIA Corporation\OptiX SDK 6.5.0\"`
### Output
```
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Users/PoAn/anaconda3/envs/fresnel/Library/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Users/PoAn/anaconda3/envs/fresnel/Library/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: C:/Users/PoAn/anaconda3/envs/fresnel/python.exe (found suitable version "3.12.7", minimum required is "3.7")
-- Found PythonLibs: C:/Users/PoAn/anaconda3/envs/fresnel/libs/python312.lib
-- pybind11::lto disabled (problems with undefined symbols for MinGW for now)
-- pybind11::thin_lto disabled (problems with undefined symbols for MinGW for now)
-- Found pybind11: C:/Users/PoAn/anaconda3/envs/fresnel/Library/include (found version "2.13.6")
-- Found pybind11: C:/Users/PoAn/anaconda3/envs/fresnel/Library/share/cmake/pybind11 C:/Users/PoAn/anaconda3/envs/fresnel/Library/include (version 2.13.6)
-- Installing fresnel python module to: C:/Users/PoAn/anaconda3/envs/fresnel/Lib/site-packages/fresnel
-- Found embree: C:/Users/PoAn/anaconda3/envs/fresnel/Library/lib/cmake/embree-4.0.0 C:/Users/PoAn/anaconda3/envs/fresnel/Library/lib/embree4.lib C:/Users/PoAn/anaconda3/envs/fresnel/Library/include
-- Found TBB: C:/Users/PoAn/anaconda3/envs/fresnel/Library/lib/cmake/TBB C:/Users/PoAn/anaconda3/envs/fresnel/Library/bin/tbb12.dll C:/Users/PoAn/anaconda3/envs/fresnel/Library/include
-- Found Qhull: C:/Users/PoAn/anaconda3/envs/fresnel/Library/lib/qhull_r.lib
-- Performing Test can_link_libqhullcpp
-- Performing Test can_link_libqhullcpp - Failed
-- Configuring done (2.0s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_OPTIX
    OptiX_DIR

-- Build files have been written to: D:/fresnel/fresnel/build
```
### Thoughts
I am suspecting that this has something to do with the test can_link_libqhullcpp, but I couldn't tell what went wrong from the associated file "FindQhull" in the CMake folder. I am not familiar with CMake so please forgive me if this is a trivial mistake. I would appreciate any suggestions you may have. Thanks!

### Versions
- Python 3.12

**Comment by joaander** [comment]
There is a typo in the documentation. The option is `ENABLE_OPTIX`, not `BUILD_OPTIX`: https://github.com/glotzerlab/fresnel/blob/a05be1ffc425ef12a881c78b1e7624e1224224f4/CMakeLists.txt#L115

That being said, I have never tested fresnel with Optix on Windows. You may encounter other problems after setting `-DENABLE_OPTIX=ON`.

You can ignore the `-- Performing Test can_link_libqhullcpp - Failed` message. fresnel builds libqhullcpp from source when it cannot link to it.
